src/sql/workbench/api/node/mainThreadAccountManagement.ts

```diff
@@ -32,6 +33,20 @@ export class MainThreadAccountManagement implements MainThreadAccountManagementS
 			this._proxy = extHostContext.getProxy(SqlExtHostContext.ExtHostAccountManagement);
 		}
 		this._toDispose = [];
+
+		this._accountManagementService.updateAccountListEvent((e: UpdateAccountListEventParams) => {
+			if (!e) {
+				return;
+			}
+
+			const providerMetadataIndex = Object.values(this._providerMetadata).findIndex((providerMetadata: sqlops.AccountProviderMetadata) => providerMetadata.id === e.providerId);
```

Comment:
Suggestion: why not just use `for (let key in this._providerMetadata)` and then check `if (this._providerMetadata[key].id === e.providerId)` notify account changed and break the loop? It seems simpler than finding index based on value, then handling separate return case and separately looking up key based on index.

Reply:
1. Removed AccountWithProviderHandle
2. Use a private dictionary _accounts in ExtHostAccountManagement to cache all accounts and corresponding provider handles.
3. getSecurityToken gets provider handle from _accounts for specified account.
4. Added / changed unit tests for getAllAccounts & getSecurityToken